aixplain/modules/finetune/hyperparameters.py

```diff
@@ -2,16 +2,30 @@
 from dataclasses_json import dataclass_json
 
 
+class SchedulerType:
+    LINEAR = "linear"
+    COSINE = "cosine"
+    COSINE_WITH_RESTARTS = "cosine_with_restarts"
+    POLYNOMIAL = "polynomial"
+    CONSTANT = "constant"
+    CONSTANT_WITH_WARMUP = "constant_with_warmup"
+    INVERSE_SQRT = "inverse_sqrt"
+    REDUCE_ON_PLATEAU = "reduce_lr_on_plateau"
+
+
 @dataclass_json
 @dataclass
 class Hyperparameters(object):
     epochs: int = 4
     train_batch_size: int = 4
     eval_batch_size: int = 4
     learning_rate: float = 2e-5
-    warmup_steps: int = 500
     generation_max_length: int = 225
     tokenizer_batch_size: int = 256
     gradient_checkpointing: bool = False
     gradient_accumulation_steps: int = 1
     max_seq_length: int = 4096
+    warmup_ratio: float = 0.0
+    warmup_steps: int = 0
+    early_stopping_patience: int = 1
+    lr_scheduler_type: SchedulerType = SchedulerType.LINEAR
```

Comment:
Does @dataclass_json converts the enumerator to string automatically?

Reply:
yes, by running the following line:
```python 
finetune = FinetuneFactory.create(str(uuid.uuid4()), dataset_list, model, hyperparameters=Hyperparameters(lr_scheduler_type=SchedulerType.CONSTANT))
```
We get the following request body:
```json
{
  "datasets": [
    {
      "datasetId": "65293795abed4d551b86baac",
      "trainPercentage": 100,
      "devPercentage": 0
    }
  ],
  "sourceModelId": "64e615671567f848804985e1",
  "parameters": {
    "hyperparameters": {
      "epochs": 4,
      "train_batch_size": 4,
      "eval_batch_size": 4,
      "learning_rate": 2e-05,
      "generation_max_length": 225,
      "tokenizer_batch_size": 256,
      "gradient_checkpointing": false,
      "gradient_accumulation_steps": 1,
      "max_seq_length": 4096,
      "warmup_ratio": 0.0,
      "warmup_steps": 0,
      "early_stopping_patience": 1,
      "lr_scheduler_type": "constant"
    }
  }
}
```

Reply:
should you include these parameters in the functional test?